examples/daal/cpp/source/k_nearest_neighbors/bf_knn_dense_batch.cpp

```diff
@@ -0,0 +1,151 @@
+NumericTablePtr testGroundTruth;
+NumericTablePtr testData;
+
+char *kchar;
+
+void trainModel();
+void testModel();
+void printResults();
+
+int main(int argc, char *argv[]) {
+  checkArguments(argc, argv, 2, &trainDatasetFileName, &testDatasetFileName);
+
+  kchar = getenv("K");
+
+  trainModel();
+  testModel();
+  printResults();
+
+  return 0;
+}
+
+void trainModel() {
+  /* Initialize FileDataSource<CSVFeatureManager> to retrieve the input data
+   * from a .csv file */
+  FileDataSource<CSVFeatureManager> trainDataSource(
+      trainDatasetFileName, DataSource::notAllocateNumericTable,
+      DataSource::doDictionaryFromContext);
+
+  /* Create Numeric Tables for training data and labels */
+  NumericTablePtr trainData(
+      new HomogenNumericTable<>(nFeatures, 0, NumericTable::doNotAllocate));
+  NumericTablePtr trainGroundTruth(
+      new HomogenNumericTable<>(1, 0, NumericTable::doNotAllocate));
+  NumericTablePtr mergedData(
+      new MergedNumericTable(trainData, trainGroundTruth));
+
+  /* Retrieve the data from the input file */
+  trainDataSource.loadDataBlock(mergedData.get());
+
+  /* Create an algorithm object to train the KD-tree based kNN model */
+  bf_knn_classification::training::Batch<> algorithm;
+
+  /* Pass the training data set and dependent values to the algorithm */
+  algorithm.input.set(classifier::training::data, trainData);
+  algorithm.input.set(classifier::training::labels, trainGroundTruth);
+
+  /* Train the KD-tree based kNN model */
+  algorithm.compute();
+
+  /* Retrieve the results of the training algorithm  */
+  trainingResult = algorithm.getResult();
+}
+
+void testModel() {
+  /* Initialize FileDataSource<CSVFeatureManager> to retrieve the test data from
+   * a .csv file */
+  FileDataSource<CSVFeatureManager> testDataSource(
+      testDatasetFileName, DataSource::notAllocateNumericTable,
+      DataSource::doDictionaryFromContext);
+
+  /* Create Numeric Tables for testing data and labels */
+  testData = NumericTablePtr(
+      new HomogenNumericTable<>(nFeatures, 0, NumericTable::doNotAllocate));
+  testGroundTruth = NumericTablePtr(
+      new HomogenNumericTable<>(1, 0, NumericTable::doNotAllocate));
+  NumericTablePtr mergedData(new MergedNumericTable(testData, testGroundTruth));
+
+  /* Retrieve the data from input file */
+  testDataSource.loadDataBlock(mergedData.get());
+
+  /* Create algorithm objects for KD-tree based kNN prediction with the default
+   * method */
+  bf_knn_classification::prediction::Batch<> algorithm;
+
+  /* Pass the testing data set and trained model to the algorithm */
+  algorithm.input.set(classifier::prediction::data, testData);
+  algorithm.input.set(classifier::prediction::model,
+                      trainingResult->get(classifier::training::model));
+  algorithm.parameter().nClasses = nClasses;
+  algorithm.parameter().k = kchar[0] - '0';
+  algorithm.parameter().voteWeights = bf_knn_classification::voteDistance;
```

Comment:
It isn't minimal example with that

Reply:
Examples was minimized. Parameters were removed